doc/reference_cluster.xml

```diff
@@ -393,7 +465,75 @@ GEOMETRYCOLLECTION(LINESTRING(6 6,7 7))
         <para>
           <xref linkend="ST_ClusterDBSCAN" />,
           <xref linkend="ST_ClusterKMeans" />,
-          <xref linkend="ST_ClusterIntersecting" />
+          <xref linkend="ST_ClusterIntersectingWin"/>,
+          <xref linkend="ST_ClusterWithinWin"/>,
+          <xref linkend="ST_ClusterIntersecting"/>
+        </para>
+      </refsection>
+
+    </refentry>
+
+  <refentry id="ST_ClusterWithinWin">
+      <refnamediv>
+        <refname>ST_ClusterWithinWin</refname>
+
+        <refpurpose>Window function that returns a cluster id for each input geometry, clustering using separation distance.</refpurpose>
+      </refnamediv>
+
+      <refsynopsisdiv>
+        <funcsynopsis>
+          <funcprototype>
+            <funcdef>integer <function>ST_ClusterWithinWin</function></funcdef>
+            <paramdef><type>geometry winset </type> <parameter>geom</parameter></paramdef>
+            <paramdef><type>float8 </type> <parameter>distance</parameter></paramdef>
+          </funcprototype>
+        </funcsynopsis>
+      </refsynopsisdiv>
+
+      <refsection>
+        <title>Description</title>
+
+        <para>ST_ClusterWithinWin is a window function that returns an integer for each input geometry, where the integer the cluster number the geometry is a member of. Clusters represent a set of input geometries separated by no more than the specified distance. (Distances are Cartesian distances in the units of the SRID.)
```

Comment:
Worth mentioning that this function is equivalent to `ST_ClusterDBSCAN(eps, minPoints:=0)` ?

Reply:
Are they fully equivalent?  In behaviour as well as performance?  If so, perhaps only one is needed?

Reply:
Yes, `ST_ClusterWithin` just calls DBSCAN.

Reply:
If it's equivalent to ST_ClusterDBSCan, then yah we should not bother adding it. One less function and then just add an addendum to the docs.

Reply:
@dbaston 
> Worth mentioning that this function is equivalent to `ST_ClusterDBSCAN(eps, minPoints:=0)` ?

Is there a reason you didn't set minPoints as default arg with value 0?

Reply:
As an API for DBSCAN, which requires both parameters, having a default value for `minPoints` didn't seem appropriate to me.

Reply:
OTOH, seems to me that there could be a faster implementation for ClusterWithin (since unlike DBSCAN it is a deterministic function which doesn't have to be interative).  So providing an specific API for it will allow a future optimized implementation, and lets users be specific about what they want to compute.

Reply:
DBSCAN with minPoints=0 is deterministic. I'm not sure what you would propose changing about it? I have no opinion on adding a new function or not, I just thought it would be good to point out the equivalency in the docs.

Reply:
(It's already special-cased, if that's what you're getting at) https://github.com/postgis/postgis/blob/1a004f40f7fb446aa84caae213d4b782ea74e209/liblwgeom/lwgeom_geos_cluster.c#L514-L520

Reply:
> It's already special-cased

Well, that's a good optimization.  And seems like it might actually be the optimal implementation for Cluster-Within?  (A bit hard to tell, since the generality makes the algorithm somewhat obscure).

Anyway, the documentation for `ST_ClusterDBSCAN` has a complex description which does not make it easy to deduce that it is also useful for a deterministic Cluster-Within functionality.  So having a separate function and doc page seems ideal.

Reply:
Yes, my suggestion was only to document the equivalence.